sklearn/cross_decomposition/pls_.py

```diff
@@ -38,7 +39,9 @@ def _nipals_twoblocks_inner_loop(X, Y, mode="A", max_iter=500, tol=1e-06,
         # 1.1 Update u: the X weights
         if mode == "B":
             if X_pinv is None:
-                X_pinv = linalg.pinv(X)   # compute once pinv(X)
+                # We use slower pinv2 (same as np.linalg.pinv) for stability
+                # reason
```

Comment:
we are, and it should be checked in test_common (but please confirm). Also I think ``check_finite`` is not available in scipy 0.9, right?